lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -179,15 +179,18 @@ def doomed?
       end
 
       def doom_reason
-        exit_code = @status.dig('lastState', 'terminated', 'exitCode')
-        last_terminated_reason = @status.dig("lastState", "terminated", "reason")
         limbo_reason = @status.dig("state", "waiting", "reason")
         limbo_message = @status.dig("state", "waiting", "message")
 
-        if last_terminated_reason == "ContainerCannotRun"
+        if @status.dig("lastState", "terminated", "reason") == "ContainerCannotRun"
           # ref: https://github.com/kubernetes/kubernetes/blob/562e721ece8a16e05c7e7d6bdd6334c910733ab2/pkg/kubelet/dockershim/docker_container.go#L353
+          exit_code = @status.dig('lastState', 'terminated', 'exitCode')
           "Failed to start (exit #{exit_code}): #{@status.dig('lastState', 'terminated', 'message')}"
+        elsif @status.dig("state", "terminated", "reason") == "ContainerCannotRun"
```

Comment:
Can you add a comment that this is a 1.7 thing?

Reply:
I couldn't find the change by poking around in the source, but I wouldn't be surprised if it was a fix / not going to change in the future. Unmanaged pods aren't restarted, so having `lastState` populated doesn't really make sense.